src/ddmd/root/outbuffer.d

```diff
@@ -417,3 +429,34 @@ struct OutBuffer
         return extractData();
     }
 }
+
+/****** copied from core.internal.string *************/
```

Comment:
Would love to find a way to centralize this code somewhere.  I've found it useful in bare-metal/no-runtime programming also.